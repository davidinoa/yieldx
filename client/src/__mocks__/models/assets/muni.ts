import {
  Asset,
  AssetAssetTypeEnum,
  AssetAssetSubtypeEnum
} from "@bondhouse/rover-universe";

const muni: Asset = {
  id: "bond",
  name: "Bond",
  description: "Bond",
  price: 105,
  ticker: "CT",
  priceDate: "2020-05-27",
  country: "US",
  assetType: AssetAssetTypeEnum.Bond,
  assetSubtype: AssetAssetSubtypeEnum.Sovereign,
  currency: "USD",
  universes: {
    liquid: true
  },
  identifiers: {
    cusip: "20030NCM1",
    isin: "muniIsin"
  },
  bond: {
    issuer: {
      id: "muniIssuer"
    },
    cleanPrice: 105,
    accruedInterest: 1,
    maturityDate: "2023-08-01",
    classifications: {
      datastreamType: "Municipals",
      datastreamSubtype: "Municipals"
    },
    instrumentType: "Bond",
    originalAmountIssued: 29620,
    amountOutstanding: 29620,
    capType: "New Filing",
    debtServiceType: "Special Tax",
    state: "CT",
    useOfProceeds: "General Purpose/Public Improvements",
    minimumDenomination: 5000,
    denominationIncrement: 5000,
    ratings: {
      snpLongTerm: {
        effectiveDate: "2015-09-28",
        rating: "AA",
        rank: 5
      }
    },
    unitedStatesTaxExempt: true,
    stateTaxExempt: true
  },
  analytics: {
    date: "2020-05-27",
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    yield: 1,
    duration: 2
  },
  allocation: {
    country: {
      us: 100
    },
    currency: {
      usd: 100
    },
    maturity: {
      threeToFiveYears: 100
    },
    rating: {
      aa: 100
    },
    region: {
      unitedStates: 100
    }
  }
};

export default muni;
