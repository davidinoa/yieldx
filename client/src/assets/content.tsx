export const ToolTipContent = {
  riskComparison:
    "Compare the InPaaS portfolio risk to similar yielding securities",
  expenseComparison:
    "Compare the InPaaS portfolio fees to similar yielding securities.",
  targetYield: "Target pre-tax yield of the portfolio.",
  targetIncome: "Target pre-tax monthly income earned from the portfolio.",
  CurrentYield: `Current annual yield of the portfolio.
  Fund yields are net of fund manager fees and expenses.`,
  CurrentIncome: "Current pre-tax monthly income of the portfolio.",
  DistributionIncome: "Current pre-tax monthly income of the portfolio",
  ExpenseRatio:
    "Market-value-weighted average of the expense ratio of all funds within the portfolio.",
  maxDrawdown:
    "The maximum loss from peak to trough in cumulative total returns of the portfolio over the last 5 years.",
  Yield:
    "Annual investment income as a percentage of the market value of the portfolio.",
  Duration:
    "Market-value-weighted average of the effective durations of the individual bonds within the portfolio.",
  Maturity:
    "Market-value-weighted average years to maturity of the individual bonds within the portfolio.",
  Spread:
    "Market-value-weighted average of the credit spreads within the portfolio.",
  risk: "Annualized volatility of the portfolio returns.",
  SectorAllocation:
    "Percentage breakdown of portfolio allocation to various fixed income sectors.",
  EsgScore:
    "Environmental, Social, and Govenerence scores powered by Clarity.AI.",
  YieldContribution: {
    Bonds: "Percentage that each bond contributes towards the total yield.",
    Funds: "Percentage that each fund contributes towards the total yield."
  },
  PortfolioGrowth: `Forecasted growth of the portfolio based on target yield at creation.
     Shaded region represents 30th and 70th percentile outcome.`,
  Ratings:
    "Percentage breakdown of the portfolio allocation to each S&P credit rating bucket, such that plus (+) and minus (-) ratings are grouped in the same bucket",
  BondRatings:
    "Percentage breakdown of the bond portfolio allocation to each S&P credit rating",
  FundAllocation:
    "Percentage of the portfolio invested in Funds (ETFs & Closed-End Funds)",
  BondAllocation: "Percentage of the portfolio invested in Bonds",
  Positions: {
    Default: "Portfolio constituents",
    Next10:
      "Alternative bond suggestions should liquidity prevent execution of any single bond or user desires to further exclude specific bonds"
  }
};
