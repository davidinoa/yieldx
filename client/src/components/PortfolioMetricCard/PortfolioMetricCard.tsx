import React, { useState } from "react";
import { Col } from "react-bootstrap";
import Tooltip from "../../elements/Tooltip/Tooltip";
import { ReactComponent as Edit } from "../../assets/svg/edit.svg";
import { CardHeader, CardRoot, EditLink } from "./PortfolioMetricCard.styles";
import { ColBreakPoints } from "../../theme/breakpoints.style";
import CardValue from "./CardValue/CardValue";
import ToggleValue from "./ToggleValue/ToggleValue";
import { Spinner } from "../../elements/Spinner/Spinner";

export type ValueFormat = "%" | "$" | "bps";
type ToggleValueOptions = {
  label: string;
  number?: number;
  toggleLabel?: ValueFormat | string;
  format: ValueFormat;
  tooltipContent?: string;
  helperText?: string;
};

interface ToggleProps extends ColBreakPoints {
  defaultToggleLabel?: ValueFormat | string;
  value?: ToggleValueOptions;
  secondValue?: ToggleValueOptions;
  toggle?: JSX.Element;
  children?: never;
  onEdit?: () => void;
  label?: never;
  tooltipContent?: never;
}

interface CustomContentProps extends ColBreakPoints {
  children: JSX.Element | JSX.Element[];
  label: string;
  tooltipContent?: string;
  toggle?: never;
  defaultToggleLabel?: never;
  value?: never;
  secondValue?: never;
  onEdit?: never;
}

export default function PortfolioMetricCard(
  props: ToggleProps | CustomContentProps
) {
  const {
    toggle,
    children,
    onEdit,
    value,
    secondValue,
    defaultToggleLabel,
    label,
    tooltipContent,
    xs
  } = props;
  const [currentToggleLabel, setCurrentToggleLabel] = useState(
    defaultToggleLabel
  );
  let currentValue: ToggleValueOptions | undefined = value;
  if (
    secondValue !== undefined &&
    currentToggleLabel === secondValue.toggleLabel
  )
    currentValue = secondValue;
  return (
    <Col xs={xs} className="mb-1" data-testid="Portfolio-Metric-Card">
      <CardRoot>
        <CardHeader className="d-flex align-items-center">
          {currentValue !== undefined ? currentValue.label : label}
          {tooltipContent && <Tooltip content={tooltipContent} />}
          {!tooltipContent && currentValue && currentValue.tooltipContent && (
            <Tooltip content={currentValue.tooltipContent} />
          )}
          {value &&
            currentValue &&
            currentValue.toggleLabel &&
            value.toggleLabel &&
            secondValue &&
            secondValue.toggleLabel && (
              <ToggleValue
                className="ml-auto"
                onToggle={() => {
                  if (currentValue?.toggleLabel === secondValue?.toggleLabel)
                    setCurrentToggleLabel(value.toggleLabel);
                  else setCurrentToggleLabel(secondValue?.toggleLabel);
                }}
                switchToggleLabels={
                  currentValue.toggleLabel === "%" ||
                  currentValue.toggleLabel === "$"
                    ? ["%", "$"]
                    : undefined
                }
                current={currentValue.toggleLabel}
              />
            )}
          {!currentValue?.toggleLabel && toggle}
          {onEdit && (
            <EditLink data-testid="Card-EditLink" onClick={onEdit}>
              <Edit /> Edit
            </EditLink>
          )}
        </CardHeader>
        {children !== undefined && children}

        {currentValue !== undefined && currentValue.number === undefined && (
          <Spinner />
        )}
        {currentValue !== undefined && currentValue.number !== undefined && (
          <CardValue
            value={{
              number: currentValue.number,
              format: currentValue.format,
              helperText: currentValue.helperText
            }}
          />
        )}
      </CardRoot>
    </Col>
  );
}
