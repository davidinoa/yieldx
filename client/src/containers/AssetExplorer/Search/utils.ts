interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
