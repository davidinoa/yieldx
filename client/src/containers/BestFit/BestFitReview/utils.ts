import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
