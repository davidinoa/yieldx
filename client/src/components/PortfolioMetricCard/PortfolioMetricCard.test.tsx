import React from "react";
import userEvent from "@testing-library/user-event";
import { render, screen } from "test/utils";
import { ToolTipContent } from "assets/content";
import PortfolioMetricCard from "./PortfolioMetricCard";

describe("PortfolioMetricCard", () => {
  it("Should render card with title and children", () => {
    const { asFragment } = render(
      <PortfolioMetricCard label="Risk comparison">
        <h1>mock content</h1>
      </PortfolioMetricCard>
    );
    expect(screen.queryByText(/risk comparison/i)).toBeInTheDocument();
    expect(screen.queryByText(/mock content/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render card with title, children, and tooltip", () => {
    const { asFragment } = render(
      <PortfolioMetricCard
        label="Risk comparison"
        tooltipContent="mock tooltip"
      >
        <h1>mock content</h1>
      </PortfolioMetricCard>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render with single value", () => {
    const { asFragment } = render(
      <PortfolioMetricCard
        xs={6}
        value={{
          tooltipContent: ToolTipContent.targetYield,
          label: "Yield",
          number: 5.0,
          format: "%"
        }}
      />
    );
    expect(screen.queryByText(/yield/i)).toBeInTheDocument();
    expect(screen.queryByText(/5.00/i)).toBeInTheDocument();
    expect(screen.queryByText(/%/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render with two values and text button to switch", async () => {
    const { asFragment } = render(
      <PortfolioMetricCard
        xs={6}
        value={{
          tooltipContent: ToolTipContent.targetYield,
          label: "Income",
          number: 525,
          format: "$",
          toggleLabel: "Annual",
          helperText: "monthly"
        }}
        secondValue={{
          tooltipContent: ToolTipContent.targetIncome,
          label: "Income",
          number: 525 * 12,
          format: "$",
          toggleLabel: "Monthly",
          helperText: "yearly"
        }}
      />
    );
    expect(screen.queryByText(/Income/i)).toBeInTheDocument();
    expect(screen.queryByText(/\$525/i)).toBeInTheDocument();
    expect(screen.queryByText(/\/monthly/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
    const textButton = await screen.findByText(/annual/i);
    expect(textButton).toBeInTheDocument();
    userEvent.click(textButton);

    expect(screen.queryByText(/monthly/i)).toBeInTheDocument();
    expect(screen.queryByText(/\/yearly/i)).toBeInTheDocument();
    expect(screen.queryByText(/\$6,300/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
