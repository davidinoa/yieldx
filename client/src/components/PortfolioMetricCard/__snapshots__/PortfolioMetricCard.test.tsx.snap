// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioMetricCard Should render card with title and children 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c1 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

@media (min-width:1600px) {
  .c0 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c1 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c1 > svg {
    width: 1rem;
    height: 1rem;
  }
}

<div
    class="mb-1 col"
    data-testid="Portfolio-Metric-Card"
  >
    <div
      class="c0"
    >
      <div
        class="c1 d-flex align-items-center"
      >
        Risk comparison
      </div>
      <h1>
        mock content
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioMetricCard Should render card with title, children, and tooltip 1`] = `
<DocumentFragment>
  .c2 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c0 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c1 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

@media (min-width:1600px) {
  .c2 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c0 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c1 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c1 > svg {
    width: 1rem;
    height: 1rem;
  }
}

<div
    class="mb-1 col"
    data-testid="Portfolio-Metric-Card"
  >
    <div
      class="c0"
    >
      <div
        class="c1 d-flex align-items-center"
      >
        Risk comparison
        <i
          class="c2"
        >
          <svg>
            info.svg
          </svg>
        </i>
      </div>
      <h1>
        mock content
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioMetricCard Should render with single value 1`] = `
<DocumentFragment>
  .c2 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c0 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c1 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c3 {
  line-height: normal;
  font-size: 1.5rem;
}

.c3 sup {
  font-size: 1rem;
  padding: 0.25rem;
}

.c3 span {
  color: var(--subtext-3);
  font-size: 12px;
  line-height: 15px;
}

.c3 .allocation-separator {
  height: 23px;
  width: 2px;
  background: var(--border-new);
}

@media (min-width:1600px) {
  .c2 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c0 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c1 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c1 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c3 {
    font-size: 1.95rem;
  }
}

@media (min-width:1600px) {
  .c3 sup {
    font-size: 1.15rem;
  }
}

<div
    class="mb-1 col-6"
    data-testid="Portfolio-Metric-Card"
  >
    <div
      class="c0"
    >
      <div
        class="c1 d-flex align-items-center"
      >
        Yield
        <i
          class="c2"
        >
          <svg>
            info.svg
          </svg>
        </i>
      </div>
      <div
        class="c3"
      >
        5.00
        <sup>
          %
        </sup>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioMetricCard Should render with two values and text button to switch 1`] = `
<DocumentFragment>
  .c2 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c0 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c1 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c4 {
  line-height: normal;
  font-size: 1.5rem;
}

.c4 sup {
  font-size: 1rem;
  padding: 0.25rem;
}

.c4 span {
  color: var(--subtext-3);
  font-size: 12px;
  line-height: 15px;
}

.c4 .allocation-separator {
  height: 23px;
  width: 2px;
  background: var(--border-new);
}

.c3 {
  margin-left: auto;
  color: var(--lightest);
  font-size: 0.8rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  padding: 0;
  background: none;
  border: none;
}

@media (min-width:1600px) {
  .c2 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c0 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c1 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c1 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c4 {
    font-size: 1.95rem;
  }
}

@media (min-width:1600px) {
  .c4 sup {
    font-size: 1.15rem;
  }
}

<div
    class="mb-1 col-6"
    data-testid="Portfolio-Metric-Card"
  >
    <div
      class="c0"
    >
      <div
        class="c1 d-flex align-items-center"
      >
        Income
        <i
          class="c2"
        >
          <svg>
            info.svg
          </svg>
        </i>
        <button
          class="c3"
          type="button"
        >
          Annual
        </button>
      </div>
      <div
        class="c4"
      >
        $525
        <sup>
          .00
        </sup>
        <span>
          /monthly
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioMetricCard Should render with two values and text button to switch 2`] = `
<DocumentFragment>
  .c2 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c0 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c1 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c4 {
  line-height: normal;
  font-size: 1.5rem;
}

.c4 sup {
  font-size: 1rem;
  padding: 0.25rem;
}

.c4 span {
  color: var(--subtext-3);
  font-size: 12px;
  line-height: 15px;
}

.c4 .allocation-separator {
  height: 23px;
  width: 2px;
  background: var(--border-new);
}

.c3 {
  margin-left: auto;
  color: var(--lightest);
  font-size: 0.8rem;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  padding: 0;
  background: none;
  border: none;
}

@media (min-width:1600px) {
  .c2 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c0 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c1 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c1 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c4 {
    font-size: 1.95rem;
  }
}

@media (min-width:1600px) {
  .c4 sup {
    font-size: 1.15rem;
  }
}

<div
    class="mb-1 col-6"
    data-testid="Portfolio-Metric-Card"
  >
    <div
      class="c0"
    >
      <div
        class="c1 d-flex align-items-center"
      >
        Income
        <i
          class="c2"
        >
          <svg>
            info.svg
          </svg>
        </i>
        <button
          class="c3"
          type="button"
        >
          Monthly
        </button>
      </div>
      <div
        class="c4"
      >
        $6,300
        <sup>
          .00
        </sup>
        <span>
          /yearly
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
